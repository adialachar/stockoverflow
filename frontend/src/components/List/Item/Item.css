.item__container {
  border-radius: 15px;
  padding: 10px 0;
  margin-top: 10px;
}

.item__value {
  font-family: "Ubuntu", sans-serif;
  color: #FFFFFF;
}