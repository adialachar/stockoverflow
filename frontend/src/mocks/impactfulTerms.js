export const impactfulTerms = [
  'term1',
  'term2',
  'term3',
  'term4',
  'term5',
  'term6'
]