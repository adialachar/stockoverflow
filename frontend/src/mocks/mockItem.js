export const mockItemProps = {
  index: 0,
  score: 0,
  value: 'word'
}