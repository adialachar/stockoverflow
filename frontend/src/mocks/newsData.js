export const newsData = [
  {
    title: "Sample title 1",
    sentiment: "+",
    description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
  },
  {
    title: "Sample title 2",
    sentiment: "-",
    description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
  },
  {
    title: "Sample title 3",
    sentiment: "-",
    description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
  },
  {
    title: "Sample title 4",
    sentiment: "+",
    description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
  }
]