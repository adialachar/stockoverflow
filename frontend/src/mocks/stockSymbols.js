export const stockSymbols = [
  "AAPL",
  "GOOGL",
  "TRI",
  "AMAZN",
  "BCS",
  "SBUX"
]