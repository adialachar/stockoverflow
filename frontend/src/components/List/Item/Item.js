import React from 'react';
import './Item.css';

function Item(props) {
  const adjustedBackgroundColor = `rgba(73, 99, 239, ${1 - ((props.index + 1) * 0.1)})`;

  return (
    <div className="item__container" style={{ backgroundColor: adjustedBackgroundColor }}>
      <div className="item__value">
        {props.value}
      </div>
    </div>
  );
}

export default Item;