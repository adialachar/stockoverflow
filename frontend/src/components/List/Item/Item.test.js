import React from 'react';
import { render } from '@testing-library/react';
import Item from './Item';
import { mockItemProps } from '../../../mocks/mockItem';

describe("Item", () => {
  it("renders", () => {
    const { baseElement } = render(<Item { ...mockItemProps }/>);
    expect(baseElement).toMatchSnapshot();
  })
});