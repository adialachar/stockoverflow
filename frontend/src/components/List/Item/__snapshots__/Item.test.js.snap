// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders 1`] = `
<body>
  <div>
    <div
      class="item__container"
      style="background-color: rgba(73, 99, 239, 0.9);"
    >
      <div
        class="item__value"
      >
        word
      </div>
    </div>
  </div>
</body>
`;
